sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -514,18 +514,27 @@ export class BatchingReceiverLite {
     };
 
     abortSignalCleanupFunction = checkAndRegisterWithAbortSignal((err) => {
+      if (receiver.drain) {
+        // If a drain is already in process and we cancel, the link state may be out of sync
+        // with remote. Reset the link so that we will have fresh start.
+        receiver.close();
+        return;
+      }
       reject(err);
     }, args.abortSignal);
 
-    logger.verbose(
-      `${loggingPrefix} Adding credit for receiving ${args.maxMessageCount} messages.`
-    );
-
     // By adding credit here, we let the service know that at max we can handle `maxMessageCount`
     // number of messages concurrently. We will return the user an array of messages that can
     // be of size upto maxMessageCount. Then the user needs to accordingly dispose
     // (complete/abandon/defer/deadletter) the messages from the array.
-    receiver.addCredit(args.maxMessageCount);
+    const creditToAdd = args.maxMessageCount - receiver.credit;
```

Comment:
Doesn't look wrong, but when is this case happening?
If there were already credits, wouldn't the previous receive used those?

Reply:
For example, in the case where user aborts before receiver.credit (which is local to client) is consumed and updated, then user calls `receiveMessages()` again.  The same link is used since it's ready. It still has the old credits

Reply:
```ts
  while (true) {
    console.log("Trying to receive messages...");
    try{
    const messages = await receiver.receiveMessages(2, {
      maxWaitTimeInMs: 3 * 1000,
      abortSignal: AbortController.timeout(3 * 1000 - 500),
    });
    } catch (e:any) {
      console.log(e);
      if (e.code === "AbortError") {
        console.log("user aborted operation. Retrying...");
      }
    }
  };
```
Here, user can abort before we wait for the max wait time. I don't think it's good to do that, but there might be real scenario where this happens, like user interactive app where things are paused then resumed.